jlm/hls/backend/rhls2firrtl/base-hls.hpp

```diff
@@ -72,63 +74,75 @@ class BaseHLS
   static std::string
   get_base_file_name(const llvm::RvsdgModule & rm);
 
-  std::vector<jlm::rvsdg::RegionArgument *>
-  get_mem_resps(const llvm::lambda::node * lambda)
+  /**
+   * Extracts all region arguments of the given kernel that represent memory responses.
+   * They can provide multiple values within a single execution of the region.
+   * @param lambda the lambda node holding the hls kernel
+   * @return the arguments that represent memory responses
+   */
+  std::vector<rvsdg::RegionArgument *>
+  get_mem_resps(const llvm::lambda::node & lambda)
   {
-    std::vector<jlm::rvsdg::RegionArgument *> mem_resps;
-    for (size_t i = 0; i < lambda->subregion()->narguments(); ++i)
+    std::vector<rvsdg::RegionArgument *> mem_resps;
+    for (auto arg : lambda.subregion()->Arguments())
     {
-      auto arg = lambda->subregion()->argument(i);
-      if (dynamic_cast<const jlm::hls::bundletype *>(&arg->type()))
-      {
-        mem_resps.push_back(lambda->subregion()->argument(i));
-      }
+      if (rvsdg::is<bundletype>(arg->type()))
+        mem_resps.push_back(arg);
     }
     return mem_resps;
   }
 
+  /**
+   * Extracts all region results of the given kernel that represent memory requests.
+   * They can take multiple values within a single execution of the region.
+   * @param lambda the lambda node holding the hls kernel
+   * @return the results that represent memory requests
+   */
   std::vector<rvsdg::RegionResult *>
-  get_mem_reqs(const llvm::lambda::node * lambda)
+  get_mem_reqs(const llvm::lambda::node & lambda)
```

Comment:
GetMemoryRequestResults()

Reply:
If we use the term kernel for the HLS lambda then it could be named
GetKernelMemoryRequest()

Reply:
Left for later PR